pkg/controller/infrastructure/actuator_reconcile.go

```diff
@@ -117,12 +120,50 @@ func (a *actuator) applyNATGateway(ctx context.Context, ironcoreClient client.Cl
 		},
 	}
 
+	if natConfig := config.NATConfig; natConfig != nil {
+		minPortsPerNetworkInterface := natConfig.PortsPerNetworkInterface
+		if minPortsPerNetworkInterface != nil {
+			natGateway.Spec.PortsPerNetworkInterface = minPortsPerNetworkInterface
+		}
+		if maxAvailablePorts := natConfig.MaxAvailablePorts; minPortsPerNetworkInterface != nil && maxAvailablePorts != nil {
+			if nodeCIDR := cluster.Shoot.Spec.Networking.Nodes; nodeCIDR != nil {
+				_, ipv4Net, err := net.ParseCIDR(*nodeCIDR)
+				if err != nil {
+					return nil, fmt.Errorf("failed to parse node cidr %s: %w", *nodeCIDR, err)
+				}
+
+				// determines how many IP addresses reside within nodeCIDR.
+				// The first and the last IPs are NOT excluded.
+				// see reference https://github.com/cilium/cilium/blob/main/pkg/ip/ip.go#L27
+				subnet, size := ipv4Net.Mask.Size()
+				amount := big.NewInt(0).Sub(big.NewInt(2).Exp(big.NewInt(2), big.NewInt(int64(size-subnet)), nil), big.NewInt(0))
+				maxPorts := big.NewInt(int64(*maxAvailablePorts))
+				ports := big.NewInt(0).Div(maxPorts, amount)
+
+				if ports.Int64() < int64(*minPortsPerNetworkInterface) {
+					natGateway.Spec.PortsPerNetworkInterface = minPortsPerNetworkInterface
+				} else {
+					natGateway.Spec.PortsPerNetworkInterface = ptr.To(previousPowOf2(int32(ports.Int64())))
+				}
```

Comment:
I think we should have check for final `natGateway.Spec.PortsPerNetworkInterface`  value is less than `amount` which is nothing but available ports from `nodeCIDR`

Reply:
That means,  if the computed ports are smaller than min value configured, do we want to keep min value or we want keep the computed value ? @afritzler/@balpert89   what is your opinion ?

Reply:
Added validations to check its always power of 2, greater than zero and always less than max ports